File: ejerc.diccionarios/Ejercicio 4/Ejercicio4.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class FormatoFecha {
    public static void main(String[] args) {
        // Crear un HashMap para relacionar los números de mes con su nombre en español
        Map<String, String> meses = new HashMap<>();
        meses.put("01", "enero");
        meses.put("02", "febrero");
        meses.put("03", "marzo");
        meses.put("04", "abril");
        meses.put("05", "mayo");
        meses.put("06", "junio");
        meses.put("07", "julio");
        meses.put("08", "agosto");
        meses.put("09", "septiembre");
        meses.put("10", "octubre");
        meses.put("11", "noviembre");
        meses.put("12", "diciembre");

        // Crear un objeto Scanner para leer datos desde la consola
        Scanner sc = new Scanner(System.in);

        // Solicitar al usuario que ingrese la fecha en formato dd/mm/aaaa
        System.out.print("Ingrese una fecha (dd/mm/aaaa): ");
        String fecha = sc.nextLine().trim();  // .trim() elimina espacios extra

        // Dividir la fecha en partes: día, mes y año
        String[] partes = fecha.split("/");

        // Verificar que la fecha tenga tres partes y que el mes sea válido
        if (partes.length == 3 && meses.containsKey(partes[1])) {
            // Convertir el día a entero para quitar ceros a la izquierda
            int dia = Integer.parseInt(partes[0]);
            // Imprimir la fecha en formato "día de mes de año"
            System.out.println(dia + " de " + meses.get(partes[1]) + " de " + partes[2]);
        } else {
            // Mensaje de error si el mes no es válido o el formato está incorrecto
            System.out.println("El mes ingresado no es válido.");
        }

        // Cerrar el objeto Scanner
        sc.close();
    }
}

File: ejerc.diccionarios/Ejercicio 4/Ejercicio4.py
# Diccionario que relaciona los números de mes (en formato '01', '02', etc.)
# con su nombre correspondiente en español
meses = {
    '01': 'enero',
    '02': 'febrero',
    '03': 'marzo',
    '04': 'abril',
    '05': 'mayo',
    '06': 'junio',
    '07': 'julio',
    '08': 'agosto',
    '09': 'septiembre',
    '10': 'octubre',
    '11': 'noviembre',
    '12': 'diciembre'
}

# Solicitar al usuario que ingrese una fecha en formato dd/mm/aaaa
# .strip() elimina espacios en blanco al inicio y final
fecha = input("Ingrese una fecha (dd/mm/aaaa): ").strip()

# Dividir la fecha en tres partes: día, mes y año
# split('/') separa el texto usando el carácter '/' como delimitador
dia, mes, anio = fecha.split('/')

# Verificar si el mes ingresado existe en el diccionario
if mes in meses:
    # Mostrar la fecha con el nombre del mes
    # int(dia) convierte el día en número entero para evitar ceros a la izquierda
    print(f"{int(dia)} de {meses[mes]} de {anio}")
else:
    # Mensaje de error si el mes no es válido
    print("El mes ingresado no es válido.")
